### DISCUSSION TITLE
Is it possible to restart an Android activity ?

### DISCUSSION THREAD

**Discussion by EricBeetsOfficial-Opuscope** [BabylonJS/BabylonNative#876] [Development]
Hi there,

I don't know if there is a bug somewhere or if I am in a wrong way:
I am trying to use a custom `BabylonView` into a `Fragment` to add it into a _intented_ `AppCompatActivity`, everything is fine, I can go back to the `MainActivity` (the `Fragment` and `BabylonView` call the destructions functions, however the `protected void finalize` is not yet called ..).
And then, I try to relaunch this second activity (containing the custom `BabylonView`), and the app crashes (in the function `Java_BabylonNative_Wrapper_renderFrame`).
It's occurs only when the AR mode is enabled.
I can understand that some resources are not released (while the cpp pointers are not), but I don't understand the way to go (I tried to force the `finishEngine` without success ..)

I hope my description is understandable (sorry if it's not, feel free to say it !)

Thanks in advance,
Regards,
Eric

**Comment by CedricGuillemet** [comment]
Hi Eric,

Can you repro this behavior with Android Playground App? Or can you give us a link to your test app?
Please favor babylon js forum for questions/bugs (https://forum.babylonjs.com/). This place is more for feature discussion.
